core/src/main/scala/org/apache/spark/CacheManager.scala

```diff
@@ -49,7 +49,15 @@ private[spark] class CacheManager(blockManager: BlockManager) extends Logging {
           .getInputMetricsForReadMethod(inputMetrics.readMethod)
         existingMetrics.incBytesRead(inputMetrics.bytesRead)
 
-        val iter = blockResult.data.asInstanceOf[Iterator[T]]
+        val buf = blockResult.data.toArray
+        if (rdd.sparkContext.getConf.getBoolean("spark.rdd.remoteblock.cache", false)) {
```

Comment:
By doing this in default, we might cache too many unnecessary remote blocks. It should cost a lot on memory usage.